python/cuml/dask/preprocessing/encoders.py

```diff
@@ -0,0 +1,191 @@
+from cuml.preprocessing.encoders import OneHotEncoder as cumlOHE
+
+
+class OneHotEncoder(BaseEstimator, DelayedTransformMixin,
+                    DelayedInverseTransformMixin):
+    """
+    Encode categorical features as a one-hot numeric array.
+    The input to this transformer should be a dask_cuDF.DataFrame of ints or
+    strings, denoting the values taken on by categorical (discrete) features.
+    The features are encoded using a one-hot (aka 'one-of-K' or 'dummy')
+    encoding scheme. This creates a binary column for each category and
+    returns a sparse matrix or dense array (depending on the ``sparse``
+    parameter).
+    By default, the encoder derives the categories based on the unique values
+    in each feature. Alternatively, you can also specify the `categories`
+    manually.
+
+    Parameters
+    ----------
+    categories : 'auto' a dask.array or a dask_cudf.DataFrame, default='auto'
+        Categories (unique values) per feature:
+        - 'auto' : Determine categories automatically from the training data.
+        - DataFrame : ``categories[col]`` holds the categories expected in the
+          feature col.
+    drop : 'first', None or a dict, default=None
+        Specifies a methodology to use to drop one of the categories per
+        feature. This is useful in situations where perfectly collinear
+        features cause problems, such as when feeding the resulting data
+        into a neural network or an unregularized regression.
+        - None : retain all features (the default).
+        - 'first' : drop the first category in each feature. If only one
+          category is present, the feature will be dropped entirely.
+        - Dict : ``drop[col]`` is the category in feature col that
+          should be dropped.
+    sparse : bool, default=False
+        This feature was deactivated and will give an exception when True.
+        The reason is because sparse matrix are not fully supported by cupy
+        yet, causing incorrect values when computing one hot encodings.
+        See https://github.com/cupy/cupy/issues/3223
+    dtype : number type, default=np.float
+        Desired datatype of transform's output.
+    handle_unknown : {'error', 'ignore'}, default='error'
+        Whether to raise an error or ignore if an unknown categorical feature
+        is present during transform (default is to raise). When this parameter
+        is set to 'ignore' and an unknown category is encountered during
+        transform, the resulting one-hot encoded columns for this feature
+        will be all zeros. In the inverse transform, an unknown category
+        will be denoted as None.
+    """
+
+    def __init__(self, client=None, verbose=False, **kwargs):
+        super(OneHotEncoder, self).__init__(client=client,
+                                            verbose=verbose,
+                                            **kwargs)
+
+    @property
+    def categories_(self):
```

Comment:
This shouldn't be needed. At the end of fit when the local model is saved, and save it as an instance variable of the class, just to follow how other estimators do it

Reply:
Good point! I changed it according to the other estimators.